contract.md

```diff
@@ -12,29 +19,52 @@ like to work on, as part of your project (for example, other things related to D
+
+Something outside of this class that I would like the chance to work on would be to get the application side of my project up and running. So far through the HMC CS curriculum, I haven't gained very much experience writing projects that have a polished front-end result that the user can interact with. This is something that I have been interested in learning how to create but haven't gained any experience learning inside the classroom. So, I am really looking forward to connecting my DSLs to a front-end application that I can create and design. 
+
 ## Time management plan
 
 _How do you plan to set aside time outside of class, to work on the project? Are there
 intermediate milestones that you can create, to help you make consistent progress?_
 
-## Teamwork plan
+Before I go into a rough outline of a timeline I plan to follow for this project, I go over strategies I will use to maximize my productivity outside of the classroom. Having the ability to work in class is a great way to brainstorm with others and get help when stuck. I plan on using this resource to my advantage, but I will also make sure that my time spent working outside of class is equally productive. My main fear when working on my own is getting stuck and having all of my progress slowed to a halt. To avoid this, I plan on making my time spent in the classroom as useful as possible by creating a list of questions or topics I want to talk with others about before every class. 
+
+I also know that for me to maximize my efficiency when working on this project I should thoroughly plan everything out so that I can foresee any roadblocks or challenges ahead of time. This will help me make sure that I am attacking my project in a way that allows me to achieve the greatest possible end result, and it also allows me to get an idea of what I need to do and how long everything will take me to complete.
+
+Here's a rough outline of project milestones by week:
+
+Week of 3/26: Create a detailed outline of the entire project including what libraries, tools, and files need to be used/created 
 
-_If you plan to work with someone else on the project, what is your plan for
-collaborating? What part(s) do you think you might do together? What part(s) are you
-considering doing separately? How will you hold one another accountable to make regular
-progress?_
+Week of 4/2: Write DSL #1, user --> icalendar
+
+Week of 4/9: Write DSL #2, icalendar --> external app
+
+Week of 4/16: Get external application running
+
+Week of 4/23: Wrap up any loose ends
```

Comment:
If we use similar libraries for our projects, I would be happy to help debug during class time!

Your timeline looks good to me - the only part that may take longer than expected is your 2nd DSL, which seems to be more nebulous to plan at the moment and a bit more vulnerable to bugs that come up during implementation. Another thing to consider is creating an ideal input and output for the simplest case as part of your plans to motivate how your language might be designed.